### ISSUE TITLE
Expand Usability Features

### ISSUE THREAD

**Issue by kings177** [HigherOrderCO/HVM1#416]
- [x] Add error handling and reporting for IO operations
- [ ] Implement utility functions for common tasks:
  - [x] #418
  - [x] #419
  - [x] #420
  - [x] #421

**Comment by enricozb** [comment]
Error handling / reporting was done: https://github.com/HigherOrderCO/HVM/pull/407

**Comment by developedby** [comment]
Before implementing any other specific file system IO, I think it'd be better to add a shell subprocess call. 
That way we have all these functions now, including a lot of stuff not included here.
Not ideal since we don't get nice errors but it fills the need of most users already.

**Comment by enricozb** [comment]
Yeah I was imagining subprocess could be a solid substitution for these in the meantime

**Comment by VictorTaelin** [comment]
HigherOrderCO's repositories are being reorganized: HVM1 and HVM2 swapped repository names, so that each holds its own code and history. We are closing every open issue and pull request as part of the move: if this one is still relevant, please re-open it on the matching repository, https://github.com/HigherOrderCO/HVM1 or https://github.com/HigherOrderCO/HVM2.
